### PR TITLE
Set SGX execution through `WASM_VM` env. variable 

### PR DESCRIPTION
**Pull request by csegarragonz**
In this PR I change the way we indicate that we want to run a function with SGX. Even though it is not semantically correct, the environment variable indicating what `WASM_VM` we use takes now three possible values: `wavm` (default), `wamr`, and `sgx`.

This means that wen using the CLI we don't need to pass any extra parameters, just set the variable beforehand:

```bash
export WASM_VM="sgx"

# Codegen and run function inside enclave
inv codegen demo hello
inv run demo hello

export WASM_VM="wavm"

# Codegen and run function with WAVM
inv codegen demo hello
inv run demo hello
```

This also means that we can get rid of the references to SGX in the faabric message (faasm/faabric#229).

### CODE REVIEW

File: faasmcli/faasmcli/tasks/codegen.py [hunk 1]
```diff
@@ -14,7 +13,7 @@
     join(FAASM_RUNTIME_ROOT, "lib", "fake", "libfakeLibB.so"),
 ]
 
-WAMR_WHITELISTED_FUNCS = [
+WAMR_ALLOWED_FUNCS = [
```
Comment (csegarragonz) [inline faasmcli/faasmcli/tasks/codegen.py:16]: Kind of unrelated, but following our `master->main` re-factor, I thought I would also change this whilst I was amending this file (I don't think we use whitelist/blacklist anywhere else other than sgx code which will get re-factored now anyway).

Comment (Shillaker) [reply faasmcli/faasmcli/tasks/codegen.py:16]: Nice, yeah let's keep chipping away at this.

File: tests/test/codegen/test_machine_code_generator.cpp [hunk 2]
```diff
@@ -112,20 +112,15 @@ TEST_CASE_METHOD(CodegenTestFixture,
...
-        SECTION("SGX")
-        {
-            objectFileA = "/tmp/obj/demo/hello/function.aot.sgx";
-            objectFileB = "/tmp/obj/demo/echo/function.aot.sgx";
-            msgA.set_issgx(true);
-            msgB.set_issgx(true);
-        }
+    SECTION("SGX codegen")
+    {
+        conf.wasmVm = "sgx";
```
Comment (Shillaker) [inline tests/test/codegen/test_machine_code_generator.cpp:121]: Can you double check that there's a `FaasmConfigTestFixture` somewhere in the class hierarchy for this test? We just want to make sure that the config is reset at the end of each test.

**Review by Shillaker — APPROVED** [review]
Nice, much cleaner. Just a small comment about checking a test.

File: tests/test/codegen/test_machine_code_generator.cpp (hunk 2 above)
Comment (csegarragonz) [reply tests/test/codegen/test_machine_code_generator.cpp:121]: There is an `S3TestFixture` that resets the config :+1:

**Pull request merged**